pkg/agent/agent_windows.go

```diff
@@ -173,3 +174,19 @@ func (i *Initializer) initHostNetworkFlows() error {
 	}
 	return nil
 }
+
+// setupTunnelInterfaceOptions sets local_ip option of tunnel interface.
+// On windows, the local_ip option is needed to be configured to help OVS set right source IP
+// for encap packets.
+func (i *Initializer) setupTunnelInterfaceOptions(ifName string) error {
```

Comment:
Could we save the IP to InterfaceConfig.IP of TunnelInterface, and reset it only if it should be changed?
And when first creating the tunnel port, we can set the interface option together. I think tunnel port creation already supports options.

Just want to keep all port creation/options following the same workflow.

Reply:
Hi @jianjuns ,

- It's good to save `local_ip` in config. But I think mabe use `TunnelInterfaceConfig` is better. Because the `local_ip` is only used for tunnel interface, same as `remote_ip` we alreay have. So do you think if I can add `LocalIP` in `TunnelInterfaceConfig`?
```
type TunnelInterfaceConfig struct {
   	// IP address of the local Node.
	LocalIP net.IP                      <<<<<<<<<<<<<<<<<<<<<< new field.
	// IP address of the remote Node.
	RemoteIP net.IP
}
```

- Actually, CreateTunnelPort/CreateTunnelPortExt both don't receive `option` param. So how do you think if I add `local_ip` param in CreateTunnelPortExt for passing the local option?

Reply:
Ok. A localIP argument Sounds good to me.